api/envoy/extensions/original_ip_detection/common/v3/original_ip_detection.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+
+package envoy.extensions.original_ip_detection.common.v3;
+
+import "envoy/type/v3/http_status.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.original_ip_detection.common.v3";
+option java_outer_classname = "OriginalIpDetectionProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Common original IP detection types]
+
+// Common configuration for IP detection extensions.
+//
+// [#next-free-field: 6]
+message OriginalIPDetectionCommonConfig {
+  // Is the detected address trusted (e.g.: can it be used to determine if this is an internal
+  // request).
+  bool allow_trusted_address_checks = 1;
```

Comment:
Isn't the default for this true for the default xff extension? I'm probably missing something.

Reply:
actually, this is ignored by XFF so we should probably drop the common config:

```
  // Common configuration across detection extensions.
  common.v3.OriginalIPDetectionCommonConfig common_config = 2;
  ```
  
  from its config until we want to either support overriding `allow_trusted_address_checks` and/or rejecting requests for which detection fail.
  
  I'll drop it for now.

Reply:
Hmm, there's an additional thing that's confusing about this extension: this config object isn't the right place to configure the default case (`xff_num_trusted_hops` should be changed in the HCM, for the default case).

So I'll add some comments referring to the HCM for the default case and explain that this config is for when you declare your own list of extensions (per Matt's comment above of using a repated TypedExtensionConfig list) and you override the default extension.